scripts/link-checker.js

```diff
@@ -227,7 +227,7 @@ async function checkLinks() {
 
         let matchesHash = true;
         if (hash) {
-          matchesHash = !!matchingPage?.ids.includes(hash);
+          matchesHash = !!matchingPage?.ids.some((id) => id.includes(hash));
```

Comment:
Since it's not checking that the ID is exact I think an explanatory comment might be warranted, especially one that explains that this reproduces the behavior already present in GitHub.

Reply:
Done in https://github.com/w3c/aria-practices/pull/2812/commits/56d93dbc56487d3dfbe11c948a5b63f993dda70a